ksqldb-api-client/src/test/java/io/confluent/ksql/api/client/integration/ClientIntegrationTest.java

```diff
@@ -1357,11 +1361,33 @@ private static void addObjectToKsqlArray(final KsqlArray array, final Object val
       array.add(SqlTimeTypes.formatDate((Date) value));
     } else if (value instanceof Time) {
       array.add(SqlTimeTypes.formatTime((Time) value));
+    } else if (value instanceof byte[]) {
+      array.add(serializeVertX3CompatibleByte((byte[]) value));
     } else {
       array.add(value);
     }
   }
 
+  /**
+   * VertX 4 changed to serialize using Base64 without padding but our server
```

Comment:
I gave that a lot of thought but I don't think there should actually be an issue here (at least, the VertX folk seem to agree). The only reason this is a problem is because we're doing a schema-less `String` comparison on what really should be a `byte[]` type. Any Base64 decoder will decode both values identically (`AQ==` is the same as `AQ`), meaning that if you're using the values as a byte[] (which you should be) your client will be fine.